File: mtgSearch/src/main/java/com/dbottillo/mtgsearchfree/presenter/CardsMemoryStorage.java
package com.dbottillo.mtgsearchfree.presenter;

import com.dbottillo.mtgsearchfree.model.CardsBucket;

public final class CardsMemoryStorage {

    public CardsMemoryStorage() {

    }

    private CardsBucket bucket = null;

    private int[] favourites = null;

    public CardsBucket getBucket() {
        return bucket;
    }

    public void setBucket(CardsBucket bucket) {
        this.bucket = bucket;
    }

    public int[] getFavourites() {
        return favourites;
    }

    public void setFavourites(int[] favourites) {
        this.favourites = favourites;
    }

    public boolean isBucketType(String value) {
        return !(bucket == null || !bucket.getKey().equals(value));
    }

    public void invalidate() {
        bucket = null;
    }
}

File: mtgSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/presenter/CardsMemoryStorage.kt
package com.dbottillo.mtgsearchfree.presenter

import com.dbottillo.mtgsearchfree.resources.MTGCard
import java.util.*

object CardsMemoryStorage {
    var currentKey: String = ""
    var cards: List<MTGCard> = ArrayList()

    fun isValid(key: String): Boolean {
        if (currentKey.equals(key) && cards.size > 0) {
            return true;
        }
        return false;
    }

    fun update(newKey: String, newCards: List<MTGCard>) {
        currentKey = newKey
        cards = newCards
    }
}